src/msg/simple/Accepter.cc

```diff
@@ -304,6 +305,13 @@ void *Accepter::entry()
   pfd[0].events = POLLIN | POLLERR | POLLNVAL | POLLHUP;
   pfd[1].fd = shutdown_rd_fd;
   pfd[1].events = POLLIN | POLLERR | POLLNVAL | POLLHUP;
+
+  if (msgr->cct->get_module_type() == CEPH_ENTITY_TYPE_OSD) {
+    std::stringstream ss;
+  	ss << "Accepter thread " << (void *)pthread_self();
+    hb = msgr->cct->get_heartbeat_map()->add_worker(ss.str(), pthread_self());
```

Comment:
Ah, so the ```hb``` is a class member now and that's good. But I think instead of setting it in Accepter::entry based on ```msgr->cct->get_module_type()```, we should have ceph_osd.cc (or maybe OSD.cc, whichever is appropriate) set Accepter::hb. We don't really want the messenger to be explicitly aware of the daemon types.

Although I suppose we could do this later after porting it into the AsyncMessenger as well so we have a common interface.

Reply:
Before modification, We  break out the Accepter::entry when there are serious errors in network communication. I do not think we need to mark ourself heathy again when connections come again.
Setting hb to osd.cc(maybe) looks like appropriate. I will modify it.